sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringOperations.scala

```diff
@@ -275,6 +276,31 @@ case class StringLength(child: Expression) extends UnaryExpression with ExpectsI
 }
 
 /**
+ * A function that return soundex code of the given string expression.
+ */
+case class SoundEx(child: Expression) extends UnaryExpression with ExpectsInputTypes {
+  override def dataType: DataType = StringType
+
+  override def inputTypes: Seq[DataType] = Seq(StringType)
+
+  protected override def nullSafeEval(string: Any): Any = SoundEx(child)
```

Comment:
Let's move the logic from the `eval` to `nullSafeEval`, and remove the null checking,